### PR TITLE
Add documentation to k6 exit codes definitions

### PR DESCRIPTION
**Pull request by oleiade**
## What?

This Pull Request adds Godoc documentation comment to k6 exit codes.

I've added my context, but if reviewers have more about specific exit codes they wish to add, that would be the perfect opportunity.

## Why?

Having recently been in the situation of interpreting the status code (on-call) for debugging purposes, I found that in some instances, the reason and context for each status code could have been more straightforward. Hence, I added documentation to each of them, albeit probably not complete and not having as much context as possible.

## Checklist

- [X] I have performed a self-review of my code.
- [X] I have added tests for my changes.
- [X] I have run linter locally (`make ci-like-lint`) and all checks pass.
- [X] I have run tests locally (`make tests`) and all tests pass.
- [X] I have commented on my code, particularly in hard-to-understand areas.

### CODE REVIEW

**Review by olegbespalov — APPROVED** [review]
Great stuff :heart: Thanks for doing this :bow: 

To be more precise, one possible improvement could be to explicitly name `k6's REST API` in the places where it's mentioned. What do you think?

**Comment by oleiade** [comment]
I'm sorry, I'm not sure to understand what you're suggesting. What changes do you have in mind? 🙇🏻

File: errext/exitcodes/codes.go [hunk 1]
```diff
@@ -8,17 +8,49 @@ type ExitCode uint8
...
+
+	// InvalidConfig indicates an invalid configuration.
+	InvalidConfig ExitCode = 104
+
+	// ExternalAbort indicates the test was aborted by an external signal
+	// (e.g. SIGINT, SIGTERM, etc.) and should be considered aborted rather
+	// than a failure.
+	ExternalAbort ExitCode = 105
+
+	// CannotStartRESTAPI indicates the REST API server could not be started.
```
Comment (olegbespalov) [inline errext/exitcodes/codes.go:43]:
```suggestion
	// CannotStartRESTAPI indicates the k6's REST API server could not be started.
```

File: errext/exitcodes/codes.go [hunk 2]
```diff
@@ -8,17 +8,49 @@ type ExitCode uint8
...
+	SetupTimeout ExitCode = 100
+
+	// TeardownTimeout indicates the execution of the test teardown function timed out.
+	TeardownTimeout ExitCode = 101
+
+	// GenericTimeout indicates a timeout with an unspecified reason.
+	GenericTimeout ExitCode = 102 // TODO: remove?
+
+	// ScriptStoppedFromRESTAPI indicates the execution has been
+	// stopped by a call to the REST API.
```
Comment (olegbespalov) [inline errext/exitcodes/codes.go:32]:
```suggestion
	// stopped by a call to the k6's REST API.
```

**Review by olegbespalov — DISMISSED** [review]
@oleiade I've added direct suggestion feel free to accept or drop them

**Review by imiric — APPROVED** [review]
Save for a couple of cases like `ExternalAbort`, most of these comments seem redundant to me, as they repeat in prose what is already self-explained by the constant names. But if you and others find this helpful, I'm not against it.

**Comment by olegbespalov** [comment]
@imiric they might seem redundant, but in fact, such comments also help go documentation properly parse and put such constants into the proper place. E.g. right now the section is empty https://pkg.go.dev/go.k6.io/k6@v0.45.1/errext/exitcodes#pkg-constants

For sure the consts themself, are still presented and self-explanatory, but they also provide a better UX with a bit of cost.

**Comment by oleiade** [comment]
I agree that they might come up as redundant, but they're actually the result of me struggling to match k6 exit codes to their meaning "easily".

I'd still like to merge them, and eventually even improve them 👍

**Review by olegbespalov — APPROVED** [review]

**Review by mstoykov — APPROVED** [review]
I have the same opinion as @imiric, but I guess if others find it useful -let's do it.

**Pull request merged**